src/GridFieldConfigurablePaginator.php

```diff
@@ -64,10 +67,21 @@ public function __construct($itemsPerPage = null, $pageSizes = null)
      * Get the total number of records in the list
      *
      * @return int
+     * @deprecated 4.1.1 Use getTotalItems() instead.
      */
     public function getTotalRecords()
     {
-        return (int) $this->getGridField()->getList()->count();
+        Deprecation::notice('4.1.1', 'Use getTotalItems() instead.');
+        return $this->getTotalItems();
+    }
+
+    public function getTotalItems(): int
+    {
+        if (!$this->haveCheckedCount) {
+            $this->totalItems = (int) $this->getGridField()->getList()->count();
+            $this->haveCheckedCount = true;
+        }
```

Comment:
Unlike in the parent class, this is relatively likely to be called at various points, so we need to check if we've counted.

Can't just rely on the value of `totalItems` because `0` is the default and we don't want to keep counting the list if it's legitimately empty (more db queries)